store/entity-store.html

```diff
@@ -294,62 +292,33 @@
 				this._initContainer(this._listeners, entityId, cacheKey, new Set()).delete(listener);
 			}.bind(this));
 		},
-		_handleCachePriming: function (href, token, response, linkHeader) {
+		_handleCachePriming: function (href, token, response) {
+			var linkHeaderValues = response.headers && headers.headers.get('Link');
+			if (!linkHeaderValues) {
+				return response.json();
+			}
+
 			var cachePrimers = window.D2L.Siren.EntityStore
-				.parseLinkHeader(linkHeader)
+				.parseLinkHeader(linkHeaderValues)
 				.filter(function(link) {
 					return link.rel.indexOf('https://api.brightspace.com/rels/cache-primer') !== -1;
 				});
-			if (cachePrimers.length !== 1) {
+
+			if (cachePrimers.length === 0) {
 				return response.json();
 			}
-			var self = this;
-			return this._getToken(token).then(function(resolved){
-				var cacheKey = resolved.cacheKey;
-				return new Promise(function(resolve) {
-					var headers = new Headers();
-					var cachePrimerHref = cachePrimers[0].href;
-					resolved.tokenValue && headers.set('Authorization', 'Bearer ' + resolved.tokenValue)
-						  && headers.set('pragma', 'no-cache')
-						  && headers.set('cache-control', 'no-cache');
-					window.d2lfetch.fetch(cachePrimerHref, {
-						headers: headers
-					})
-					.then(function(cacheResponse){
-						if (!cacheResponse.ok) {
-							throw new Error("exit and proceed");
-						}
-						return cacheResponse.json();
-					})
-					.then(function(data){
-						var entity = window.D2L.Hypermedia.Siren.Parse(data);
-						//prime up that cache
-						const subEntities = entity.getSubEntitiesByRel('canonical');
-						for( var subEntity of subEntities) {
-							const subEntityId = subEntity.getLinkByRel && subEntity.getLinkByRel('self') && subEntity.getLinkByRel('self').href || '';
-							// don't process if this is the entity we are about to prime, or if it has no self-link
-							if( subEntityId === '' || subEntityId === href) {
-								continue;
-							}
-							const key = subEntityId.toLowerCase();
-							// since its canonical, we'll remove the context in which we found it,
-							// cache it as _the_ entity for this href
-							delete subEntity.rel;
-							self._initContainer(self._store, key, cacheKey);
-							self._store.get(cacheKey).set(key, {
-								status: '',
-								entity: subEntity,
-								request: null
-							});
-						}
-						return resolve(response.json());
-					})
-					.catch(function(err){
-					//if it fails.. meh, just go on
-						return resolve(response.json())
-					});
-				});
-			});
+
+			function returnJson() {
+				return response.json();
+			}
+
+			return Promise
+				.all(cachePrimers.map(function(cachePrimer) {
+					return this
+						.fetch(cachePrimer.href, token, true)
```

Comment:
so obviously this isn't doing exactly the same, but I figured we could just re-use the fetch/expand behaviour? Maybe make `expand` more selective (find what's relying on it and require a `canonical` class or a `canonical-item` rel? Only other different might be a `nocache` option or the like to avoid adding the cacheprimer entity to the cache